newswires/test/db/FingerpostWireEntrySpec.scala

```diff
@@ -320,4 +321,53 @@ class FingerpostWireEntrySpec extends AnyFlatSpec with Matchers with models {
 
     query.statement should include("ORDER BY fm.ingested_at ASC")
   }
+
+  ignore should "format and output the full sql query" in {
+    // This is meant as a handy way to get the full sql for a query from the Scala parameter list.
+    // The parameters are just an example, change them for whatever you want to test.
+    // Handy to get the full query with parameters inserted, so it can be pasted into a `psql` session,
+    // for example for testing changes, or inspecting the query planner with an EXPLAIN query.
+
+    val sp = SearchParams(
+      text = None
+    )
+    val queryParams = QueryParams(
+      searchParams = sp,
+      savedSearchParamList =
+        SearchPresets.get("all-world").get ++ SearchPresets.get("all-uk").get,
+      maybeSearchTerm = None,
+      maybeBeforeId = None,
+      maybeSinceId = None,
+      pageSize = 20
+    )
+
+    val whereClause = buildWhereClause(
+      queryParams.searchParams,
+      queryParams.savedSearchParamList,
+      maybeBeforeId = queryParams.maybeBeforeId,
+      maybeSinceId = queryParams.maybeSinceId.map(_.sinceId)
+    )
+
+    val query = buildSearchQuery(queryParams, whereClause)
+
+    // Replace the parameter placeholders with the corresponding values
+    // WARNING
+    // This is a textbook example of how to get hit by a SQL injection attack - this should be used
+    // for troubleshooting only, and if any part of the query comes from an untrusted source,
+    // you really need to scan the output to check that no part is attempting to perform malicious actions.
+    val fmted = query.parameters.foldLeft(
+      query.statement.replaceAll("\\?\\?", "?")
+    )((statement, param) => {
+      val matcher = "\\?(?![|?])"
+      param match {
+        case p: Int    => statement.replaceFirst(matcher, s"$p ")
+        case p: String => statement.replaceFirst(matcher, s"'$p' ")
```

Comment:
just spotted that this will fail if the param is the name of a column, because that doesn't want to be placed inside `'`s.

we are currently doing that in some queries (I think here? https://github.com/guardian/newswires/blob/d00ab66faa21d9f8a046d00154673f71cdd6d490/newswires/app/db/FingerpostWireEntry.scala#L288-L293).

not sure if it's worth trying to account for this edge case in the code, but thought i'd flag it as something to be aware of for future travellers :)